R/outbreak_model.R

```diff
@@ -72,13 +42,33 @@
 #'   quarantine = FALSE
 #' )
 #' out
-outbreak_model <- function(initial_cases = NULL, prop_ascertain = NULL,
-                           cap_max_days = NULL, cap_cases = NULL,
-                           r0_isolated = NULL, r0_community = NULL,
-                           r0_asymptomatic = NULL, disp_isolated = NULL,
-                           disp_community = NULL, disp_asymptomatic = NULL,
-                           k, onset_to_isolation, incubation_period,
-                           prop_asymptomatic = NULL, quarantine = FALSE) {
+outbreak_model <- function(initial_cases,
+                           r0_community, r0_isolated, r0_asymptomatic,
+                           disp_community, disp_isolated, disp_asymptomatic,
+                           incubation_period, k,
+                           onset_to_isolation,
+                           prop_ascertain, prop_asymptomatic,
+                           cap_max_days, cap_cases,
+                           quarantine = FALSE) {
+
+  check_outbreak_input(
```

Comment:
what would make the most sense to me here is calling this function within the environment or some such, rather than having manually pass everything by name.

Reply:
Could you provide (pseudo)code how it should be called within the environment?

Reply:
so i think you'd rewrite check_outbreak_input to only have func as an argument, then call it in the environment. the enclosing function and its arguments should then be in the search path for symbols, so would be available as appropriate inside check_outbreak_input

Reply:
Looks fine to me. Curiousity item tho: re the environmental capture as `args` in check_outbreak_input - is that actually required to make this work? or more something to please some of the CI / check / etc?

Reply:
Assigning to an object as a list, `args`, and then extracting for each check was the only way I thought of doing it. Also maybe it's safest approach to be sure it's a variable from the parent environment rather than the current one, although for the simple use here I doubt it matters.